zdict/dictionary.py

```diff
@@ -17,18 +17,27 @@ class DictBase(metaclass=abc.ABCMeta):
     def __init__(self, args):
         self.args = args
         self.db = db
-        self.db.connect()
 
-        for req in self.REQUIRED_TABLE:
-            if not req.table_exists():
-                req.create_table()
+        try:
+            self.db.connect()
+        except AttributeError as e:
+            print('Cannot connect db: {}'.format(e))
```

Comment:
How about setting `self.db` to `None` also?
Then we can just check the value of `self.db` in `__del__`.